File: social-flow-backend/common/libraries/node/auth/password.js
/**
 * Password hashing and verification with bcrypt
 */

const bcrypt = require('bcrypt');
const config = require('./config');

module.exports = {
  hashPassword: async (password) => {
    return await bcrypt.hash(password, config.security.saltRounds);
  },

  verifyPassword: async (password, hash) => {
    return await bcrypt.compare(password, hash);
  },
};

File: social-flow-backend/shared/libraries/python/security/password.py
# password.py
import bcrypt


class PasswordHasher:
    """
    Secure password hashing and verification.
    Uses bcrypt (adaptive hashing).
    """

    @staticmethod
    def hash_password(password: str) -> str:
        return bcrypt.hashpw(password.encode(), bcrypt.gensalt()).decode()

    @staticmethod
    def verify_password(password: str, hashed: str) -> bool:
        return bcrypt.checkpw(password.encode(), hashed.encode())
